Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -76,27 +76,32 @@ internal class ConnectionPool {
 
     /// Initializes the pool using the provided `ConnectionString` and options.
     internal init(from connString: ConnectionString, options: ClientOptions?) throws {
+        // validate option before we bother creating pool, so we don't have to destroy it on error.
+        if let maxPoolSize = options?.maxPoolSize {
+            guard maxPoolSize > 0 && maxPoolSize <= UInt32.max else {
+                throw InvalidArgumentError(
+                    message: "Invalid maxPoolSize \(maxPoolSize): must be between 1 and \(UInt32.max)"
+                )
+            }
+        }
+
         let pool: OpaquePointer = try connString.withMongocURI { uriPtr in
             guard let pool = mongoc_client_pool_new(uriPtr) else {
                 throw InternalError(message: "Failed to initialize libmongoc client pool")
             }
             return pool
         }
 
-        guard mongoc_client_pool_set_error_api(pool, MONGOC_ERROR_API_VERSION_2) else {
-            throw InternalError(message: "Could not configure error handling on client pool")
-        }
+        self.state = .open(pool: pool)
 
         if let maxPoolSize = options?.maxPoolSize {
-            guard maxPoolSize > 0 && maxPoolSize <= UInt32.max else {
-                throw InvalidArgumentError(
-                    message: "Invalid maxPoolSize \(maxPoolSize): must be between 1 and \(UInt32.max)"
-                )
-            }
             mongoc_client_pool_max_size(pool, UInt32(maxPoolSize))
         }
 
-        self.state = .open(pool: pool)
+        guard mongoc_client_pool_set_error_api(pool, MONGOC_ERROR_API_VERSION_2) else {
```

Comment:
I made this change to let us get away with not having to clean up the pool here due to the concerns noted in my previous comment...
this method only returns false if the error API is invalid. given that libmongoc will never break things, and we control the version now, I don't think it's possible that we would pass in an invalid error API version here. it was of more concern in the past because users controlled their libmongoc version and could plausibly try to use the driver with an old version without this API (though I'm not sure if the driver would have compiled with a version that old anyway since we use a lot of newer API methods.)